lib/config.rb

```diff
@@ -48,6 +49,14 @@ def self.load_files(*sources)
 
     config.add_source!(Sources::EnvSource.new(ENV)) if Config.use_env
 
+    if defined?(::Rails::Railtie) && Config.use_rails_credentials
+      if Rails.version < '7.1'
```

Comment:
Because the version numbers are represented as strings here, this comparison can lead to unexpected results. For example, if `Rails.version` is `10.0.0`, then `'10.0.0' < '7.1' == true`

To compare version numbers correctly, I think we need to either use something like

```ruby
if [Rails::VERSION::MAJOR, Rails::VERSION::MINOR] < [7, 1]
```

or

```ruby
if Gem::Version.new(Rails.version) < Gem::Version.new('7.1')
```